Modules/_pickle.c

```diff
@@ -4546,21 +4578,23 @@ _pickle_Pickler_clear_memo_impl(PicklerObject *self)
 
 _pickle.Pickler.dump
 
+  cls: defining_class
   obj: object
   /
 
 Write a pickled representation of the given object to the open file.
 [clinic start generated code]*/
 
 static PyObject *
-_pickle_Pickler_dump(PicklerObject *self, PyObject *obj)
-/*[clinic end generated code: output=87ecad1261e02ac7 input=552eb1c0f52260d9]*/
+_pickle_Pickler_dump_impl(PicklerObject *self, PyTypeObject *cls,
+                          PyObject *obj)
+/*[clinic end generated code: output=952cf7f68b1445bb input=f949d84151983594]*/
 {
+    PickleState *st = _Pickle_GetStateByClass(cls);
```

Comment:
`Pickler` has `Py_TPFLAGS_BASETYPE` set, is it safe to use `_Pickle_GetStateByClass`?

Reply:
Good catch; no that definitely sounds unsafe.

Reply:
Right, I missed that.